File: tests/HartreeSinc/init_density.jl
function init_density( num_gaussian, position::Array{Float64}, coef::Array{Float64},
                       exponent::Array{Float64}, size, scaling, grid)

    weight = prod(scaling)

    total_size = prod(size)
    density = zeros(total_size)
    potential = zeros(total_size)

    for igauss = 1:num_gaussian
        exponent2 = exponent[igauss]^2
        for i_x = 1:size[1]
            for i_y = 1:size[2]
                for i_z = 1:size[3]
                    x = grid[1,i_x] - position[1,igauss]
                    y = grid[2,i_y] - position[2,igauss]
                    z = grid[3,i_z] - position[3,igauss]
                    r2 = x*x + y*y + z*z
                    # density
                    idx = (i_x-1)*size[2]*size[3] + (i_y-1)*size[3] + (i_z-1) + 1
                    density[idx]   += coef[igauss]*(exponent2/pi)^1.5 * exp(-exponent2*r2)*sqrt(weight)
                    # the potential (analytic) due to the Gaussian density
                    potential[idx] += coef[igauss]*erf(exponent[igauss]*sqrt(r2))/sqrt(r2)
                end
            end
        end
    end
    #
    return density, potential
end

File: tests/HartreeSinc/init_density.f90
SUBROUTINE init_density( num_gaussian, positions, coefs, exponents, density, potential )

  USE m_constants, ONLY : PI
  USE m_LF3d, ONLY : hh => LF3d_hh, &
                     Npoints => LF3d_Npoints, &
                     lingrid => LF3d_lingrid
  IMPLICIT NONE 
  !
  INTEGER :: num_gaussian
  REAL(8) :: positions(3,num_gaussian)
  REAL(8) :: exponents(num_gaussian), coefs(num_gaussian)
  REAL(8) :: density(Npoints)
  REAL(8) :: potential(Npoints)
  !
  REAL(8) :: weight, exponent2, r2
  REAL(8) :: r(3)
  INTEGER :: igauss, ip

  weight = hh(1)*hh(2)*hh(3)  ! FIXME use dVol ??

  density(:)   = 0.d0
  potential(:) = 0.d0
  !
  DO igauss = 1, num_gaussian
    exponent2 = exponents(igauss)**2
    DO ip = 1,Npoints
      r(:) = lingrid(:,ip) - positions(:,igauss)
      r2 = r(1)**2 + r(2)**2 + r(3)**2
      density(ip) = density(ip) + coefs(igauss)*(exponent2/pi)**1.5d0 * exp(-exponent2*r2)*sqrt(weight)
      ! the potential (analytic) due to the Gaussian density
      potential(ip) = potential(ip) + coefs(igauss)*erf(exponents(igauss)*sqrt(r2))/sqrt(r2)
    ENDDO 
  ENDDO 
END SUBROUTINE 

